### DISCUSSION TITLE
ZeroTier with SmoothWAN

### DISCUSSION THREAD

**Discussion by meassick** [SmoothWAN/SmoothWAN#100] [General & Help]
Is it possible to use ZeroTier with SmoothWan, in particular on the GL.iNet Flint router?  The functionality for Tailscale is built-in, but I have been using ZeroTier on many other devices successfully

**Comment by TalalMash** [comment]
May add next release since the OpenWRT official version is outdated and no usable LuCI app.
The problem is that the newer versions are riddled with memory leaks and are unstable and they need careful configuration.
Are you using the latest release for a good period?

**Reply by meassick** [reply]
I have been using version 1.6.5 of zerotier on openwrt routers (using Goldenorb software) with LTE modems for a couple months now and have not had issues.  However, I reboot the router nightly so I would not see memory leaks if they are there.  I also use version 1.10.1 on Windows with not issues.

I

**Comment by TalalMash** [comment]
Latest ARM official build is using a missing `libstdc++10` dependency, which is not shipped with any OWRT version from what I can tell hence why it's crashing.
Uploaded is a temporary fixed package till next version. (1.8.4-2 seems to be the last to use c++6)
[zerotier_1.8.4-2_arm_cortex-a7.ipk.zip](https://github.com/SmoothWAN/SmoothWAN/files/10690485/zerotier_1.8.4-2_arm_cortex-a7.ipk.zip)

**Comment by meassick** [comment]
No there was not, I followed the instructions given in Github: Home · mwarning/zerotier-openwrt Wiki (github.com)<https://github.com/mwarning/zerotier-openwrt/wiki> using the LuCI commands

The installation worked with no glitches.  Based on OpenWrt 19.07

Is there any issue with the OpenWRT version considering that you used it with Goldenorb?
Seems to install successfuly on a glinet SmoothWAN and up to date:
[image]<https://user-images.githubusercontent.com/96490382/217664862-4da1a2a8-8a9a-449c-ab8a-6eb0b5dba6ff.png>
You mean a LuCI app for managment?

**Reply by TalalMash** [reply]
See my comment above for a working package.

**Reply by meassick** [reply]
The package you generated installs per the Wiki.  I am able to access the router remotely as I would expect, except that the Speedify UI (from the   Status>Overview page) does not display correctly.  In the window it always says "Restart Speedify Service" even though it is connected to the service.  If at the same time, I access the router directly (not through ZeroTier), it shows the correct UI.  Any idea what the issue may be here?

**Reply by TalalMash** [reply]
Ah yes, port mapping was used on the older versions, for <0.99.8 click the Fullscreen link and add to the URL: `&wsPort=9331`

**Reply by meassick** [reply]
This works on version 0.99.8 that I have installed, but only in the fullscreen link as you indicated.  Will this fix be added to a future releases?  Also, are you planning on incorporating Zerotier like Tailscale or will it always be installation via an ipk package?

**Reply by TalalMash** [reply]
I've released 0.99.9 with these changes yesterday.
Regarding the package, that is unofficial, and I'm not sure why OpenWRT has released the official version for ARMv7 in a broken state.
Will add once they fix it.

**Reply by meassick** [reply]
The changes in 0.99.9 with port mapping for Speedify status works when using ZeroTier.  The one issue I notices is that with access through ZeroTier, the terminal window (services>terminal) would not display and just show "xxx.xxx.xxx.xxx refused to connect"  where xxx.xxx.xxx.xxx is the ip to access the router through ZeroTier

**Reply by TalalMash** [reply]
Not sure if ttyd websocket does source origin check filtering. It probably follows openwrt dropbear WAN access policy. Since cloudflared tunnel is a proxy not a tun interface.

**Reply by meassick** [reply]
As I understand it ttyd is a package for openwrt.  As I really do not know anything about openwrt, how can this be debugged to get it working?
Also, the issue is the same if I use Tailscale

**Reply by TalalMash** [reply]
@meassick Try setting interface to unspecified
![image](https://user-images.githubusercontent.com/96490382/221894919-a569bc0b-aec2-4cff-baa2-85c50cc64e84.png)

**Reply by meassick** [reply]
That works for Tailscale.  I will try it for Zerotier when I get it installed.

**Reply by TalalMash** [reply]
Should no longer be needed with latest release.
